tests/generation/test_stopping_criteria.py

```diff
@@ -107,3 +108,57 @@ def test_validate_stopping_criteria(self):
         stopping_criteria = validate_stopping_criteria(StoppingCriteriaList(), 11)
 
         self.assertEqual(len(stopping_criteria), 1)
+
+    def test_stop_string_criteria(self):
+        true_strings = [
+            "<|im_start|><|im_end|>",
+            "<|im_start|><|im_end|<|im_end|>",
+            ">><|im_start|>>stop",
+            "stop",
+            "end",
+        ]
+        false_strings = [
+            "<|im_start|><|im_end|",
+            "<|im_start|><|im_end|<|im_end|",
+            "<|im_end|><|im_start|>",
+            "<|im_end|<>stop<|im_end|",
```

Comment:
```suggestion
            "<|im_end|<>stop<|im_end|",
            "en d",
            "eNd",
```
just in case we do some space/capitalization shenanigans in the future 🙃

Reply:
Done!